modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/ValueProviderModelValidator.java

```diff
@@ -45,48 +54,111 @@
 
   @Override
   public void validate(ExtensionModel model, ProblemsReporter problemsReporter) {
+    final ReflectionCache reflectionCache = new ReflectionCache();
+    boolean isCompileTime = isCompiletime(model);
```

Comment:
the runtime already has a centralized control to decide when to invoke validators and when not. If the runtime decides it wants to run the validations, the validator should not go rogue and decide otherwise. This is just a flag which doesn't fix the problems with this algorithms